Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -708,17 +678,14 @@ ComponentInfo::buildMergeIndicesSync(const ComponentInfo &other) const {
 void ComponentInfo::checkSizes(const ComponentInfo &other) const {
   if (size() != other.size())
     failMerge("size mismatch");
-  if (!m_scanIntervals || !other.m_scanIntervals)
-    failMerge("scan intervals not defined");
 }
 
 void ComponentInfo::checkIdenticalIntervals(
-    const ComponentInfo &other, const size_t linearIndexOther,
-    const size_t linearIndexThis) const {
-  if ((*m_positions)[linearIndexThis] != (*other.m_positions)[linearIndexOther])
+    const ComponentInfo &other, const std::pair<size_t, size_t> indexOther,
+    const std::pair<size_t, size_t> indexThis) const {
+  if (this->position(indexThis) != other.position(indexOther))
```

Comment:
It is surpising that this change fixed the unit test you mentioned, given my comment above regarding iterating only `nonDetectorSize()`. However, since detectors always have the lowest indices, you may just be checking the first chunk of components here which coincidentally are detectors, but miss to compare the last chunk?